vetomint/tests/integration_test.rs

```diff
@@ -83,3 +85,594 @@ fn success_trivial_1() {
+            time: 3,
+        };
+        let response = progress(&height_info, &mut state, event);
+        assert!(response.is_empty());
+    }
+    let event = ConsensusEvent::Prevote {
+        proposal: 0,
+        round: 0,
+        signer: 3,
+        time: 3,
+    };
+    let response = progress(&height_info, &mut state, event);
+    assert_eq!(
+        response,
+        vec![ConsensusResponse::BroadcastPrecommit {
+            proposal: 0,
+            round: 0
+        }]
+    );
+
+    let timeout = height_info.consensus_params.timeout_ms as i64;
+
+    //Step3: Timeout
+    let event = ConsensusEvent::Timer { time: timeout };
+
+    let _response = progress(&height_info, &mut state, event);
+
+    for validator_index in 0..=2 {
+        let event = ConsensusEvent::NilPrecommit {
+            round: 0,
+            signer: validator_index,
+            time: timeout,
+        };
+        let response = progress(&height_info, &mut state, event);
+        assert!(response.is_empty());
+    }
+
+    let event = ConsensusEvent::NilPrecommit {
+        signer: 3,
+        round: 0,
+        time: timeout,
+    };
+
+    let response = progress(&height_info, &mut state, event);
+    assert_ne!(
+        response,
+        vec![ConsensusResponse::FinalizeBlock { proposal: 0 }]
+    );
+}
+
+#[ignore]
+#[test]
+// New round with 5/6++ prevote e.t. and not pre-committed.
+//unimplemented
+fn new_round_et_five_sixths_not_committed() {
+    let height_info = HeightInfo {
+        validators: vec![1, 1, 1, 1, 1, 1, 1],
+        this_node_index: 6,
+        timestamp: 0,
+        consensus_params: ConsensusParams { timeout_ms: 1000 },
+    };
+    let mut state = ConsensusState::new(height_info.clone());
+
+    // STEP 1: Proposal.
+    let event = ConsensusEvent::BlockProposal {
+        proposal: 0,
+        proposer: 0,
+        round: 0,
+        time: 1,
+    };
+    let response = progress(&height_info, &mut state, event);
+    assert!(response.is_empty());
+    let event = ConsensusEvent::ProposalFavor {
+        proposal: 0,
+        favor: true,
+        time: 2,
+    };
+    let response = progress(&height_info, &mut state, event);
+    assert_eq!(
+        response,
+        vec![ConsensusResponse::BroadcastPrevote {
+            proposal: 0,
+            round: 0
+        }]
+    );
+
+    let _event = ConsensusEvent::NilPrevote {
+        signer: 4,
+        round: 0,
+        time: 3,
+    };
+
+    // STEP 2: Prevote.
+    for validator_index in 0..=2 {
+        let event = ConsensusEvent::Prevote {
+            proposal: 0,
+            round: 0,
+            signer: validator_index,
+            time: 3,
+        };
+        let response = progress(&height_info, &mut state, event);
+        assert!(response.is_empty());
+    }
+    let event = ConsensusEvent::Prevote {
+        proposal: 0,
+        round: 0,
+        signer: 3,
+        time: 3,
+    };
+    let response = progress(&height_info, &mut state, event);
```

Comment:
(3+1)/7 < 5/6 
so early termination will not occur

Reply:
As I commented above,
0, 1, 2, 3 nodes prevote to proposal
node 6 favors in this proposal
node 4 prevote to nil
thus, (4+1+1)/7 < 5/6
Or, did any change occur in ProposalFavor? it seems point does not match on ProposalFavor